mob

```diff
@@ -209,157 +226,201 @@ while not os.path.exists(os.path.join(top_level, ".mobconf")):
+##
+if args.run_as_root:
+    container_ssh_dir = '/root/.ssh'
+else:
+    # Run processes in the container as root or your local user ID
     uid = os.getuid()
     gid = os.getgid()
     username = getpass.getuser()
     groupname = grp.getgrgid(gid)[0]
     docker_run.extend([
-            "--env", "EXTERNAL_UID=" + str(uid),
-            "--env", "EXTERNAL_GID=" + str(gid),
-            "--env", "EXTERNAL_USER=" + username,
-            "--env", "EXTERNAL_GROUP=" + groupname
+            '--env=EXTERNAL_UID={}'.format(str(uid)),
+            '--env=EXTERNAL_GID={}'.format(str(gid)),
+            '--env=EXTERNAL_USER={}'.format(username),
+            '--env=EXTERNAL_GROUP={}'.format(groupname)
     ])
+    # Entrypoint script will need to link/copy these files after creating the user.
+    container_ssh_dir = '/var/tmp/user/.ssh'
 
-# Add /dev/isgx if HW mode, and it is available
-# Mimicking bash checks `if [ -c /dev/isgx ]`
-# The purpose of this check is that it is possible to build in HW mode, and even
-# to run many tests, without installing sgx on the host and enabling this device in docker,
-# but sometimes you really do need it, if you want to run consensus nodes etc.
-# The friendliest thing seems to be, pass on the device if possible, maybe print a warning if not.
-if args.hw:
-    if pathlib.Path("/dev/isgx").is_char_device():
-        docker_run.extend(["--device", "/dev/isgx"])
-    else:
-        eprint("Did not find /dev/isgx on the host, skipping")
-
-# Add build environment
-build_env = [
-    "RUST_BACKTRACE=full",
-    "SGX_MODE=HW" if args.hw else "SGX_MODE=SW",
-    "IAS_MODE=PROD" if args.ias_prod else "IAS_MODE=DEV",
-]
+##
+# Set up env for builder-install or hsm-tools
+##
+if args.image == 'builder-install':
+    # Override cargo target dir so we don't conflict with a non-container build.
+    docker_run.append('--env=CARGO_TARGET_DIR={}/target/docker'.format(mount_point))
+    # Add chain id
+    docker_run.append('--env=MC_CHAIN_ID=local')
+    # Add default database url
+    docker_run.append('--env=TEST_DATABASE_URL=postgres://localhost')
+    # Add RUST_BACKTRACE
+    docker_run.append('--env=RUST_BACKTRACE=1')
+
+    # Set limits based on available cores/memory
+    jobs = get_cargo_build_jobs()
+    if jobs is not None:
+        docker_run.append("--env=CARGO_BUILD_JOBS={}".format(jobs))
 
-# Add GIT_COMMIT if present
-git_commit = get_git_commit()
-if git_commit:
-    build_env.extend([f'GIT_COMMIT={git_commit}'])
+    # set SGX_MODE options
+    if args.hw:
+        docker_run.append('--env=SGX_MODE=HW')
+        if pathlib.Path("/dev/isgx").is_char_device():
+            docker_run.append("--device=/dev/isgx")
+    else:
+        docker_run.append('--env=SGX_MODE=SW')
 
-for pair in build_env:
-    docker_run.extend(["--env", pair])
+    # set IAS_MODE options
+    if args.ias_prod:
+        docker_run.append('--env=IAS_MODE=PROD')
+    else:
+        docker_run.append('--env=IAS_MODE=DEV')
```

Comment:
Optional/DRY: Merge append calls
```suggestion
    docker_run.append('--env=IAS_MODE={}'.format('PROD' if args.ias_prod else 'DEV'))
```

Reply:
I think we should prefer more verbose syntax over condensed syntax for readability and long term maintenance.  

This is also why I also have moved to bash/shell statements on separate lines and avoid "one-liners" when possible.  I find coming back to "one-liners" after a couple of months requires a lot of decoding to figure out what was intended. 

I think it also helps those who are a bit more causal in the code base and bounce around between languages a lot.